source/models/index.py

```diff
@@ -0,0 +1,42 @@
+import os
+from outputs.exceptions import *
+
+class Index:
+    def __init__(self, table, index_value):
+        self.table = table
+        self.index_value = index_value
+
+    def serialize(self):
+        os.makedirs(self.get_path(), exist_ok = True)
+    
+    def get_path(self):
+        return os.path.join(self.table.get_path(), os.path.join("indices" , self.index_value))
+    
+    def add_primary_key(self, primary_key, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
+        old_primary_keys = set(self.get_primary_key(index_value))                
+        old_primary_keys.add(primary_key)
+        Index.__write_in_file(path = index_value_file_path, data = old_primary_keys)
```

Comment:
It is better 
```
__write_in_file(data) --> non static method 

```

`add_primary_key` shouldn't care about the file name, path , anything